### DISCUSSION TITLE
My login query doesn't work

### DISCUSSION THREAD

**Discussion by ImperatorOfIntellectualism** [urql-graphql/urql#2030] [Q&A] [answered]
My query doesn't work (it's a basic login query)

urql version & exchanges: latest

**Steps to reproduce**

Here is the full page:

import React, { useEffect, useState } from "react";
import { useMutation, useQuery } from "urql";
import styles from '../styles/Home.module.css'

interface LoginProps {}

const LOG_MUT = `query Query($login: String!, $password: String!) {
    login(login: $login, password: $password) {
      errors {
        field
        message
      }
      user {
        id
        createdAt
        updatedAt
        login
        password
      }
    }
  }`

const Login: React.FC<LoginProps> = ({}) => {
    const [user, setUser] = useState({login: '', password: ''});
    const [result,Login] = useQuery({query: LOG_MUT})
    console.log(result)
    return (
        <div className={styles.containerclass}>
  <form>
    <div>
      <h1>Login</h1>
      <hr className={styles.hrclass} />
      <label htmlFor="login"><b>Login</b></label>
      <input onChange={(event)=>{setUser({login: event.target.value, password: user.password})}} className={styles.inputclass} type="text" placeholder="Enter Login" name="login" required />
      <label htmlFor="psw"><b>Password</b></label>
      <input onChange={(event)=>{setUser({login: user.login, password: event.target.value})}} className={styles.inputclass} type="password" placeholder="Enter Password" name="psw" id="psw" required />
      <hr className={styles.hrclass} />
      <button onClick={async ()=>{Login({login: user.login, password: user.password}); if(result.data.Login.errors != null){alert(JSON.stringify(result.data.createUser.errors[0].message))} else {alert(JSON.stringify(result.data.login.user))}}} type="submit" className={styles.Loginbtn}>Login</button>
    </div>
    <div className={styles.signin}>
      <p>Don't have an account? <a className={styles.link} href="#">Sign Up here!</a>.</p>
    </div>
  </form>
</div>
      );
    };

export default Login;

**Expected behavior**

It should tell you if your user exists or not by alert()

**Actual behavior**

It says that there is no password or login, but I'm just sure that above code is wrong, because I'm used to mutations and queries are not my forte.

**Comment by kitten** [comment, marked as answer]
You haven't passed in `variables` to `useQuery`, so that explains the error you're seeing. On a side note however, since this is a stateful side effect (logging in is an action that a user takes), you'd want to use a mutation for that ❤️
